### PR TITLE
Added a NodeJS program to find occurrences of math mode.

### PR DESCRIPTION
**Pull request by notjagan**
I'm not sure as of yet if this is the format we will want the program to be in. As of now, there is a function (find_ranges) which takes a string and returns a list of places where the string is in math mode. It does not include the delimiters to start math mode. For example, passing "test $a$" to find_ranges would return [ 'a' ].

### CODE REVIEW

File: texvcjs_readinfile.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env node
```
Comment (physikerwelt) [inline texvcjs_readinfile.js]: please use strict mode

File: texvcjs_readinfile.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+var mathStart = {"\\[":"\\]","\\(":"\\)","$":"$","$$":"$$","\\begin{equation}":"\\end{equation}","\\begin{equation*}":"\\end{equation*}","\\begin{align}":"\\end{align}","\\begin{align*}":"\\end{align*}","\\begin{multline}":"\\end{multline}","\\begin{multline*}":"\\end{multline*}"};
```
Comment (physikerwelt) [inline texvcjs_readinfile.js]: I think the name is not optimal. You have a key value store where the key is the start tag and the value is the end tag.
Also newlines are cheap:-) Putting each element in a new row would help to make it more readable.

**Comment by physikerwelt** [comment]
Please
- [ ] add a test as outlined in #1 
- [ ] remove the file  `texvcjs_readinfile.js~`
- [ ] improve the commit message (https://www.mediawiki.org/wiki/Gerrit/Commit_message_guidelines)

Currently the program is quite hard to read. But the tests will help to make it more clear.

**Comment by notjagan** [comment]
I updated .jshintrc because I am using functions that call each other, which results in a used-before-declaration error.

File: .idea/.name [hunk 3]
```diff
@@ -0,0 +1 @@
+texer
```
Comment (physikerwelt) [inline .idea/.name]: Please remove all the .idea files

**Comment by physikerwelt** [comment]
Here you see which cases you miss.
https://coveralls.io/builds/5850619/source?filename=lib%2Fmathmode.js
Is it hard to develop test examples for those cases?

**Comment by physikerwelt** [comment]
@notjagan nice work.
However, please remove the .idea files.
http://stackoverflow.com/questions/3469741/remove-file-from-the-repository-but-keep-it-locally

**Pull request merged**